### PR TITLE
RFC: Reduce Build Complexity of .Net Runtimes

### PR DESCRIPTION
**Pull request by ForestEckhardt**
[Readable](https://github.com/paketo-buildpacks/rfcs/blob/simplify-dotnet-runtime/text/dotnet-core/0005-simplify-runtime-dependency.md)

### CODE REVIEW

File: text/dotnet-core/rfcs/0005-simplify-runtime-dependency.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+## Proposal
+
+Download the ASP.NET Core Runtime dependency provided by Microsoft instead of
+combining the custom built dependencies that we are currently using.
+
+## Motivation
+
+Currently, in order to get the .Net buildpack to function correctly at
+launchusing our custom dependencies, we need to build a somewhat convoluted
```
Comment (sophiewigmore) [inline text/dotnet-core/rfcs/0005-simplify-runtime-dependency.md:11]:
```suggestion
launch using our custom dependencies, we need to build a somewhat convoluted
```

**Comment by ForestEckhardt** [comment]
@fg-j I think I would like to do this as a smaller chuck of work that would make the re-ordering easier to do in the future.

**Comment by sophiewigmore** [comment]
IMO it would be nice to combine all of these concerns into one RFC

**Comment by ForestEckhardt** [comment]
I have spent a little time looking at the re-architecture and I feel like I would just need to tack an additional RFC on to the end of this RFC that is contingent on this RFC passing. I am happy to do that but I think that the purposed re-architecture that @fg-j is purposing becomes a trivial RFC that can reference this RFC and the SDK RFC once they both pass. Please let me know you would like me know what y'all are feeling and I will go with the popular vote.

**Comment by fg-j** [comment]
+1 I'd like it all to happen in this RFC. Gives us an opportunity to evaluate if there are edge cases we're missing and how we could cover them.

**Review by sophiewigmore — APPROVED** [review]

**Comment by joshuatcasey** [comment]
2022-07-26 Working Group notes: waiting for approvals

File: text/dotnet-core/0005-simplify-runtime-dependency.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+whether ASP.NET is needed at launch time.
+
+## Implementation
+
+The .NET Runtime buildpack should be archived.
+
+The ASP.NET buildpack should be archived.
+
+A new buildpack should be created, called .NET Core ASP.NET Runtime
+(`dotnet-core-aspnet-runtime`).  This mimics the name of the as it is provided
```
Comment (ForestEckhardt) [inline text/dotnet-core/0005-simplify-runtime-dependency.md:79]:
```suggestion
(`dotnet-core-aspnet-runtime`).  This mimics the name of the dependency as it is provided
```

File: text/dotnet-core/0005-simplify-runtime-dependency.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+build phase *after* application source code is compiled.
+
+## Motivation
+
+Currently, the .NET Core language family buildpack contains three component
+buildpacks that are responsible for installing parts of the .NET toolchain. The
+.NET Core SDK buildpack installs the .NET Core SDK hosted by Microsoft and
+makes it available at build time.  It's used for compiling application source
+code. The .NET Core Runtime buildpack installs a version of the .NET Core
+Runtime hosted by the Paketo dep-server. It constructs a [.NET Hive][hive] and
```
[the remark this answers is not in the document]
Comment (fg-j) [reply text/dotnet-core/0005-simplify-runtime-dependency.md:20]:
```suggestion
Runtime hosted by the Paketo dep-server. It constructs a.NET installation and
```

File: text/dotnet-core/0005-simplify-runtime-dependency.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+makes it available to framework-dependent apps at launch time. The .NET Core
+ASP.NET buildpack installs a version of ASP.NET Core that is hosted by the
+Paketo dep-server. This dependency is a *subset* of the dependency hosted by
+Microsoft. Microsoft's version contains _both_ the .NET runtime and ASP.NET
+Core. The .NET Core ASP.NET buildpack adds its dependency to the makeshift .NET
+Hive.  In other words, for most buildpack builds, the .NET Runtime and .NET
```
Comment (fg-j) [inline text/dotnet-core/0005-simplify-runtime-dependency.md:26]:
```suggestion
Core. The .NET Core ASP.NET buildpack adds its dependency to the makeshift .NET
installation.  In other words, for most buildpack builds, the .NET Runtime and .NET
```

File: text/dotnet-core/0005-simplify-runtime-dependency.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+ASP.NET buildpacks collaborate to recreate the ASP.NET Core dependency that
+Microsoft hosts and ships. Both buildpacks install their dependencies in layers
+**before** application source code is compiled. They come before the
+`dotnet-publish` buildpack in the composite buildpack's [order
+groups](https://github.com/paketo-buildpacks/dotnet-core/blob/565c719806588daaeca96e0bfd64d5743656a046/buildpack.toml).
+
+The decision to stitch together a .NET Hive was motivated by a desire to build
+images with the absolute minimal number of dependencies in them; some .NET apps
```
Comment (fg-j) [inline text/dotnet-core/0005-simplify-runtime-dependency.md:34]:
```suggestion
The decision to stitch together a .NET installation was motivated by a desire to build
images with the absolute minimal number of dependencies in them; some .NET apps
```

**Comment by fg-j** [comment]
@paketo-buildpacks/dotnet-core-maintainers Am I good to merge this and get cracking on implementation?

**Comment by dmikusa-pivotal** [comment]
2022-08-02 WG Notes:

Ready to merge.

**Pull request merged**